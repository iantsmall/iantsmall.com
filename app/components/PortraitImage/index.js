/**
 *
 * PortraitImage
 *
 */

import React, { memo } from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledContainer = styled.div`
  display: block;
  float: right;
  width: 50%;
`;
const StyledImage = styled.img`
  width: 100%;
  border-radius: 45%;
`;
function PortraitImage(props) {
  return (
    <StyledContainer>
      <StyledImage {...props} />
    </StyledContainer>
  );
}

PortraitImage.propTypes = {
  ...StyledImage.propTypes,
};

export default memo(PortraitImage);
