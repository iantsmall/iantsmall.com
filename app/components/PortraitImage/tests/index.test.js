/**
 *
 * Tests for PortraitImage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import PortraitImage from '../index';

describe('<PortraitImage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<PortraitImage />);
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<PortraitImage />);
    expect(firstChild).toMatchSnapshot();
  });
});
