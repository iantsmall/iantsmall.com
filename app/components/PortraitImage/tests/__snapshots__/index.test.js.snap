// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortraitImage /> Should render and match the snapshot 1`] = `
<div
  class="PortraitImage__StyledContainer-bnolk7-0 qQBZU"
>
  <img
    class="PortraitImage__StyledImage-bnolk7-1 lbcmjr"
  />
</div>
`;
